### PR TITLE
Concepts section

### PR DESCRIPTION
**Pull request by johndmulhausen**
Creates v1 of the concepts section

As of 7/19: Draft of Content Addressing ready for review, draft of Decentralized Storage still WIP

### CODE REVIEW

**Comment by filecorgi** [comment]
- Image optimization came back clean!
- Vuepress build was successful!

### `docs/concepts/content-addressing.md`:
- This phrase is redundant (‘S’ stands for ‘system’). Use simply “DNS”.
  ```
  ...eb3.storage is like our key-value-based DNS system, with one significant dif...
                                             ^^^^^^^^^^
  ```
- Unpaired symbol: ‘)’ seems to be missing
  ```
  ...storage space for files on the network. (For more details on that, see the D...
                                             ^
  ```
- As an alternative to the over-used intensifier ‘very’, consider replacing this phrase.
  ```
  ...two technologies, Web3.storage makes it very simple to use a decentralized s...
                                             ^^^^^^^^^^^
  ```
### `docs/concepts/decentralized-storage.md`:
- Possibly missing article found.
  ```
  ...k of distributed providers who can take deal, has strong economic incentives...
                                             ^^^^
  ```

### `docs/how-tos/retrieve.md`:
- The following link returned HTTP status code 301 (Moved Permanently):
  https://dweb.link/ipfs/bafybeidd2gyhagleh47qeg77xqndy2qy3yzn4vkxmk775bg2t5lpuy7pcu
- The following link returned HTTP status code 301 (Moved Permanently):
  https://dweb.link/ipfs/bafybeidd2gyhagleh47qeg77xqndy2qy3yzn4vkxmk775bg2t5lpuy7pcu/not-distributed.jpg
- Possibly missing comma found.
  ```
  ...t Identifier (CID)  (opens new window). First you'll need to create a Web3.S...
                                             ^^^^^
  ```
- If a new sentence starts here, add a space and start with an uppercase letter.
  ```
  ... you have a client, you can call client.get, passing in a CID string:  The g...
                                             ^^^
  ```
- An apostrophe may be missing.
  ```
  ...access to the retrieved IPFS data.  The files method returns an array of Web...
                                             ^^^^^
  ```
- If a new sentence starts here, add a space and start with an uppercase letter.
  ```
  ...is guide, we'll use the gateway at dweb.link, but you can check the list of ...
                                             ^^^^
  ```


### `docs/reference/client-library.md`:
- Possibly missing preposition found.
  ```
  ... the JavaScript client library for Web3.Storage, you must first obtain an AP...
                                             ^^^^^^^
  ```
- Possibly missing comma found.
  ```
  ...iles being uploaded. See below for more details An {options} object is a JSO...
                                             ^^^^^^^
  ```
- If a new sentence starts here, add a space and start with an uppercase letter.
  ```
  ... the iterator represent files. If entry.type == 'directory', the entry repre...
                                             ^^^^
  ```
- Possibly missing comma found.
  ```
  ...tional order. Number Type Description 1 string A string containing the CID o...
                                             ^^^^^^
  ```
- Possibly missing comma found.
  ```
  ...tional order. Number Type Description 1 string A string containing the CID o...
                                             ^^^^^^
  ```

**Comment by filecorgi** [comment]
- Image optimization came back clean!
- Vuepress build was successful!

### `docs/concepts/content-addressing.md`:
- This phrase is redundant (‘S’ stands for ‘system’). Use simply “DNS”.
  ```
  ...eb3.Storage is like our key-value-based DNS system, with one significant dif...
                                             ^^^^^^^^^^
  ```
- Check of `docs/concepts/decentralized-storage.md` came back clean!

### `docs/how-tos/retrieve.md`:
- If a new sentence starts here, add a space and start with an uppercase letter.
  ```
  ... you have a client, you can call client.get, passing in a CID string:  The g...
                                             ^^^
  ```
- An apostrophe may be missing.
  ```
  ...access to the retrieved IPFS data.  The files method returns an array of Web...
                                             ^^^^^
  ```
- If a new sentence starts here, add a space and start with an uppercase letter.
  ```
  ...is guide, we'll use the gateway at dweb.link, but you can check the list of ...
                                             ^^^^
  ```


### `docs/reference/client-library.md`:
- Possibly missing preposition found.
  ```
  ... the JavaScript client library for Web3.Storage, you must first obtain an AP...
                                             ^^^^^^^
  ```
- If a new sentence starts here, add a space and start with an uppercase letter.
  ```
  ... the iterator represent files. If entry.type == 'directory', the entry repre...
                                             ^^^^
  ```
- Possibly missing comma found.
  ```
  ...tional order. Number Type Description 1 string A string containing the CID o...
                                             ^^^^^^
  ```
- Possibly missing comma found.
  ```
  ...tional order. Number Type Description 1 string A string containing the CID o...
                                             ^^^^^^
  ```

**Comment by filecorgi** [comment]
- Image optimization came back clean!
- Vuepress build was successful!

### `docs/concepts/content-addressing.md`:
- This phrase is redundant (‘S’ stands for ‘system’). Use simply “DNS”.
  ```
  ...eb3.Storage is like our key-value-based DNS system, with one significant dif...
                                             ^^^^^^^^^^
  ```
- Check of `docs/concepts/decentralized-storage.md` came back clean!

### `docs/reference/client-library.md`:
- Possibly missing preposition found.
  ```
  ... the JavaScript client library for Web3.Storage, you must first obtain an AP...
                                             ^^^^^^^
  ```
- If a new sentence starts here, add a space and start with an uppercase letter.
  ```
  ... the iterator represent files. If entry.type == 'directory', the entry repre...
                                             ^^^^
  ```
- Possibly missing comma found.
  ```
  ...tional order. Number Type Description 1 string A string containing the CID o...
                                             ^^^^^^
  ```
- Possibly missing comma found.
  ```
  ...tional order. Number Type Description 1 string A string containing the CID o...
                                             ^^^^^^
  ```

**Review by yusefnapora — COMMENTED** [review]
This is looking good to me so far. I think it might be worth adding a sentence or two about verifiability & how you can check your downloaded data against the CID to make sure it hasn't been altered.  The verification happens under the hood by the client library, so a concept article seems like a decent place to mention it.

File: docs/reference/client-library.md [hunk 1]
```diff
@@ -50,11 +50,11 @@ Method parameters are supplied in positional order.
 | Number | Type | Description |
 | ------ | ---- | ----------- |
 | 1 | `file[]` | An iterable collection of [Files](https://developer.mozilla.org/en-US/docs/Web/API/File) to be packed into a CAR and uploaded. |
-| 2 | `{options}` | Optional. A JSON object whose properties define certain Web3.storage options and metadata about the files being uploaded. See below for more details. |
+| 2 | `{options}` | Optional. A JSON object whose properties define certain Web3.Storage options and metadata about the files being uploaded. See below for more details. |
 
 An `{options}` object is a JSON object that defines option parameters:
 
-```js
+```json
```
[the remark this answers is not in the document]
Comment (johndmulhausen) [reply docs/reference/client-library.md:57]: Ah, changed to double quotes

**Comment by johndmulhausen** [comment]
> This is looking good to me so far. I think it might be worth adding a sentence or two about verifiability & how you can check your downloaded data against the CID to make sure it hasn't been altered. The verification happens under the hood by the client library, so a concept article seems like a decent place to mention it.

Added: "It also means that your key is a signature for the data itself, ensuring _verifiability_ that the key matches the content and the content hasn't been altered."

**Comment by filecorgi** [comment]
- Image optimization came back clean!
- Vuepress build was successful!

### `docs/concepts/content-addressing.md`:
- This phrase is redundant (‘S’ stands for ‘system’). Use simply “DNS”.
  ```
  ...eb3.Storage is like our key-value-based DNS system, with one significant dif...
                                             ^^^^^^^^^^
  ```
- Check of `docs/concepts/decentralized-storage.md` came back clean!

### `docs/reference/client-library.md`:
- Possibly missing preposition found.
  ```
  ... the JavaScript client library for Web3.Storage, you must first obtain an AP...
                                             ^^^^^^^
  ```
- If a new sentence starts here, add a space and start with an uppercase letter.
  ```
  ... the iterator represent files. If entry.type == 'directory', the entry repre...
                                             ^^^^
  ```
- Possibly missing comma found.
  ```
  ...tional order. Number Type Description 1 string A string containing the CID o...
                                             ^^^^^^
  ```
- Possibly missing comma found.
  ```
  ...tional order. Number Type Description 1 string A string containing the CID o...
                                             ^^^^^^
  ```

**Comment by johndmulhausen** [comment]
Merging so I can focus on the Decentralized Storage article as all feedback is addressed and the Corgi is appeased within all reason.

**Pull request merged**